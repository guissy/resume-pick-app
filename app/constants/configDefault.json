[
  {
    "name": "核心项",
    "children": [
      {
        "name": "RN",
        "alias": ["react native", "rn", "weex"],
        "score": 1.5
      },
      {
        "name": "flutter",
        "alias": ["flutter1", "flutter2", "dart"],
        "score": 1.5
      },
      {
        "name": "h5",
        "alias": [
          "webview",
          "uni-app",
          "uniapp",
          "umi-app",
          "umijs",
          "ionic",
          "apicloud",
          "mobile",
          "hybird",
          "移动端"
        ],
        "score": 1.5
      },
      {
        "name": "react",
        "alias": ["reactjs", "redux", "antd", "ant design"],
        "score": 1.5
      },
      {
        "name": "vue",
        "alias": ["vue2", "vuejs", "vue2", "element", "vant"],
        "score": 1.5
      },
      {
        "name": "vue3",
        "score": 1.5
      },
      {
        "name": "angular",
        "alias": [
          "ng",
          "ng2",
          "angularjs",
          "angular2",
          "angular4",
          "angular5",
          "angular6",
          "angular7",
          "angular8"
        ],
        "score": 1
      },
      {
        "name": "wxapp",
        "alias": ["微信", "小程序", "taro", "mpvue", "mpvue"],
        "score": 1.5
      },
      {
        "name": "node",
        "alias": ["nodejs", "express", "koa", "koa2", "egg", "next", "pm2"],
        "score": 1.5
      },
      {
        "name": "state mgr",
        "alias": [
          "redux",
          "saga",
          "mobx",
          "vuex",
          "vuetify",
          "dva",
          "事件总线",
          "eventBus",
          "event bus",
          "mvvm"
        ],
        "score": 1
      },
      {
        "name": "es6",
        "alias": [
          "es5",
          "es6",
          "es6+",
          "es2015",
          "es2016",
          "es2017",
          "es7",
          "es8",
          "es9",
          "ecmascript",
          "原生JS",
          "原生 JS"
        ],
        "score": 0.8
      },
      {
        "name": "w3c",
        "score": 0.5,
        "alias": ["mdn"]
      },
      {
        "name": "css3",
        "alias": [
          "flex",
          "grid",
          "css3",
          "rem",
          "less",
          "sass",
          "scss",
          "css-modules",
          "css in js",
          "css-in-js",
          "styled-components",
          "jss",
          "emotion",
          "tailwind",
          "纯CSS",
          "vw",
          "vh"
        ],
        "score": 0.5
      },
      {
        "name": "keyframes",
        "alias": [
          "animate.css",
          "wow.js",
          "gsap",
          "pixijs",
          "svga",
          "createjs"
        ],
        "score": 0.8
      }
    ]
  },
  {
    "name": "进阶",
    "children": [
      {
        "name": "typescript",
        "alias": ["ts"],
        "score": 1.5
      },
      {
        "name": "hooks",
        "alias": ["hook", "hoc"],
        "score": 1.5
      },
      {
        "name": "rxjs",
        "alias": ["ReactiveX", "ngrx"],
        "score": 1.5
      },
      {
        "name": "graphql",
        "alias": ["gql"],
        "score": 1.5
      },
      {
        "name": "test",
        "calc": "max",
        "alias": [
          "mocha",
          "jasmine",
          "jest",
          "cypress",
          "puppeteer",
          "unit test"
        ],
        "score": 1.5
      },
      {
        "name": "gayhub",
        "alias": ["github.com", "gitee.com"],
        "score": 1.5
      },
      {
        "name": "lint",
        "alias": [
          "tslint",
          "eslint",
          "jslint",
          "husky",
          "prettier",
          "命名规范",
          "代码规范",
          "commit规范",
          "接口规范"
        ],
        "score": 1
      },
      {
        "name": "review",
        "alias": ["pr", "git flow", "代码审查"],
        "score": 0.5
      },
      {
        "name": "blog",
        "alias": [
          "个人博客",
          "博客地址",
          "csdn.net",
          "cnblogs.com",
          "juejin.im",
          "zhihu.com",
          "jianshu.com",
          "segmentfault.com",
          "oschina.net",
          "imooc.com",
          "51cto.com",
          "gitbooks.io"
        ],
        "score": 1.5
      }
    ]
  },
  {
    "name": "其他",
    "children": [
      {
        "name": "大厂背景",
        "alias": [
          "\\b阿里巴巴\\b",
          "\\b腾讯\\b",
          "\\b百度",
          "\\b京东\\b",
          "\\b美团点评\\b",
          "\\b字节跳动\\b"
        ],
        "score": 2.5
      },
      {
        "name": "leader",
        "alias": ["组长", "负责人", "选型", "架构设计"],
        "score": 2
      },
      {
        "name": "manager",
        "alias": ["主持会议", "分配工作", "招聘"],
        "score": 2
      },
      {
        "name": "流量",
        "alias": ["([0-9一二三四五六七八九几十百千]+)\\s*(万|亿)"],
        "score": 2
      },
      {
        "name": "百分点",
        "alias": ["(减少|降低|提升|提高)(\\D{0,2})\\d+(\\.\\d+)?%"],
        "score": 2
      },
      {
        "name": "游戏引擎",
        "alias": ["白鹭", "p2", "pomelo", "cocos2d", "egret", "重力", "粒子"],
        "score": 1.5
      },
      {
        "name": "sql",
        "alias": [
          "sqlite3",
          "sqlite",
          "mysql",
          "postgre",
          "mariadb",
          "hive",
          "redis"
        ],
        "score": 1
      },
      {
        "name": "electron",
        "alias": ["nwjs"],
        "score": 1.2
      },
      {
        "name": "mock",
        "alias": ["mockjs", "rap", "模拟", "抓取"],
        "score": 0.5
      },
      {
        "name": "ejs",
        "alias": ["hygen", "swagger"],
        "score": 0.5
      },
      {
        "name": "git",
        "score": 0.5
      },
      {
        "name": "docker",
        "score": 0.5
      },
      {
        "name": "websocket",
        "alias": ["socket", "canvas", "charles"],
        "score": 0.5
      },
      {
        "name": "微服务",
        "alias": ["monorepo", "submodule", "无服务器", "severless"],
        "score": 0.5
      },
      {
        "name": "加密",
        "alias": [
          "密文",
          "支付",
          "md5",
          "sha1",
          "签名",
          "白名单",
          "jwt",
          "x-token",
          "crypto"
        ],
        "score": 0.5
      },
      {
        "name": "重构",
        "alias": [
          "防抖",
          "去抖",
          "节流",
          "虚拟化",
          "懒加载",
          "debounce",
          "throttle",
          "代码分割",
          "分包",
          "错误收集",
          "埋点",
          "preact"
        ],
        "score": 0.5
      },
      {
        "name": "Promise",
        "alias": ["async", "await", "generator"],
        "score": 0.5
      },
      {
        "name": "自定义",
        "alias": [
          "封装",
          "公共方法",
          "二次封装",
          "造轮子",
          "简化",
          "可复用",
          "自动化"
        ],
        "score": 0.5
      },
      {
        "name": "性能调优",
        "alias": ["毫秒", "攻克"],
        "score": 0.5
      },
      {
        "name": "component",
        "alias": ["基础组件", "公共组件", "通用组件", "业务组件", "组件化"],
        "score": 0.5
      },
      {
        "name": "编程范式",
        "alias": ["函数式", "haskell", "f#"],
        "score": 0.5
      },
      {
        "name": "技术博客",
        "alias": [
          "极客",
          "掘金",
          "知乎",
          "思否",
          "stackoverflow",
          "mdn",
          "w3school",
          "博客园",
          "简书",
          "CSDN"
        ],
        "score": 0.5
      },
      {
        "name": "python",
        "score": 0.5
      },
      {
        "name": "java",
        "alias": ["springcloud", "springboot", "springmvc", "mybatis"],
        "score": 0.5
      },
      {
        "name": "middleware",
        "alias": ["nacos", "eureka", "hystrix", "gateway", "zuul", "feign"],
        "score": 0.5
      },
      {
        "name": "queue",
        "alias": ["rocketmq", "rabbitmq"],
        "score": 0.5
      },
      {
        "name": "go",
        "score": 0.5
      },
      {
        "name": "rust",
        "score": 0.5
      },
      {
        "name": "cpp",
        "score": 0.5
      },
      {
        "name": "swift",
        "score": 0.5
      },
      {
        "name": "kotlin",
        "score": 0.5
      },
      {
        "name": "scala",
        "score": 0.5
      },
      {
        "name": "mongodb",
        "alias": ["sequelize", "typeorm", "orm"],
        "score": 0.5
      },
      {
        "name": "devops",
        "alias": ["shell", "nginx", "linux", "cicd"],
        "score": 0.5
      },
      {
        "name": "vim",
        "score": 0.5
      },
      {
        "name": "markdown",
        "score": 0.5
      },
      {
        "name": "ssr",
        "score": 0.5
      },
      {
        "name": "bi",
        "score": 0.5,
        "alias": ["可视化", "大屏", "echarts", "d3", "antv"]
      },
      {
        "name": "three.js",
        "score": 0.5
      },
      {
        "name": "postman",
        "score": 0.5,
        "alias": ["charles", "fiddler"]
      },
      {
        "name": "leetcode",
        "score": 0.5,
        "alias": ["力扣", "leet code"]
      },
      {
        "name": "deno",
        "score": 0.5
      },
      {
        "name": "wasm",
        "score": 0.5,
        "alias": ["web assembly"]
      },
      {
        "name": "代表作",
        "score": 0.5,
        "alias": ["项目链接", "作品链接", "(由|从)(0|零)到(1|一)"]
      },
      {
        "name": "iframe",
        "alias": ["flash", "dreamweaver", "div+css", "v-for", "v-if"],
        "score": -1.5
      },
      {
        "name": "mvc",
        "score": -0.5
      },
      {
        "name": "ajax",
        "alias": ["ajax", "jsonp", "ie6", "ie7", "ie8"],
        "score": -1
      },
      {
        "name": "jquery",
        "alias": [
          "jq",
          "zepto",
          "knockout",
          "knockoutjs",
          "extjs",
          "backbone",
          "dojo",
          "mootools",
          "emberjs",
          "yui3",
          "prototype.js"
        ],
        "score": -0.5
      },
      {
        "name": "svn",
        "alias": [
          "requirejs",
          "gulp",
          "grunt",
          "yeo",
          "yeoman",
          "seajs",
          "phonegap",
          "jasmine"
        ],
        "score": -1
      },
      {
        "name": "积极乐观",
        "alias": ["为人和蔼", "吃苦耐劳", "上进", "了解"],
        "score": -0.5
      },
      {
        "name": "熟悉",
        "score": -0.3
      }
    ]
  }
]
