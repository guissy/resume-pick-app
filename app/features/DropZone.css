.mainBox {
  flex: 1;
  min-height: 90px;
  margin-top: 20px;
  border: 1px dashed #ddd;
  cursor: pointer;
}
.dropBox {
  width: 100%;
  height: 100%;
  padding: 30px;
  display: flex;
  justify-content: center;
  align-content: center;
  box-sizing: border-box;
}

.placeholder {
  width: 100%;
  color: #eee;
}

.textCenter {
  text-align: center;
}

.upload {
  color: #666;
}

.mainBox:hover .upload {
  color: #aaa;
}
