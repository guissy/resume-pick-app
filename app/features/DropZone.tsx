import React from 'react';
// Import the useDropzone hooks from react-dropzone
import { DropzoneProps, useDropzone } from 'react-dropzone';
import styles from './DropZone.css';

const DropZone = ({ onDrop, accept }: Partial<DropzoneProps>) => {
  // Initializing useDropzone hooks with options
  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    accept,
  });

  /*
    useDropzone hooks exposes two functions called getRootProps and getInputProps
    and also exposes isDragActive boolean
  */

  return (
    // eslint-disable-next-line react/jsx-props-no-spreading
    <div {...getRootProps()} className={styles.mainBox}>
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <input className="dropzone-input" {...getInputProps()} />
      <div className={styles.dropBox}>
        <section className={styles.placeholder}>
          {isDragActive ? (
            <p className={styles.textCenter}>文件放在这里</p>
          ) : (
            <p className={styles.textCenter}>
              {/* eslint-disable-next-line react/no-unescaped-entities */}
              拖拽文件到这里
            </p>
          )}
        </section>
      </div>
    </div>
  );
};

export default DropZone;
