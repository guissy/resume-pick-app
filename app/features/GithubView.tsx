import React from 'react';
import { GhOpt, HTMLWebview } from './type';
import styles from './ScoreList.css';

type Props = {
  url: string;
};
const GithubView: React.FC<Props> = ({ url }) => {
  const webviewRef = React.useRef<HTMLWebview | null>(null);
  const [repos, setRepos] = React.useState(NaN);
  const [contrib, setContrib] = React.useState(NaN);
  const [followers, setFollowers] = React.useState(NaN);
  // const [calendar, setCalendar] = React.useState('');
  React.useEffect(() => {
    webviewRef.current?.addEventListener('dom-ready', () => {
      webviewRef.current
        ?.executeJavaScript(
          `JSON.stringify({
            repos: parseInt(document.querySelector('.Counter')?.textContent),
            followers: parseInt(document.querySelector(".js-profile-editable-area a")?.textContent?.trim()),
            contrib: parseInt(document.querySelector('.js-yearly-contributions h2')?.textContent?.trim()),
            calendar: "document.querySelector('.js-calendar-graph-svg').outerHTML",
          })`,
          false
        )
        .then((data) => {
          const result = JSON.parse(data) as GhOpt;
          if (typeof result.repos === 'number') {
            setRepos(result.repos);
          }
          if (typeof result.contrib === 'number') {
            setContrib(result.contrib);
          }
          if (typeof result.followers === 'number') {
            setFollowers(result.followers);
          }
          // setCalendar(result.calendar);
        });
    });
  }, [webviewRef]);
  const tail = url.split('://').pop();
  const [host, path] = tail?.split('/') || [];
  const urlOk = `https://${host}/${path.split('?').shift()}`;
  return url.includes('github') ? (
    <div>
      <webview ref={webviewRef} src={urlOk} style={{ display: 'none' }} />
      {repos >= 0 && (
        <>
          <span className={styles.subName}>repo</span>
          <span className={styles.gained}>{repos}</span>
          <br />
        </>
      )}
      {followers >= 0 && (
        <>
          <span className={styles.subName}>followers</span>
          <span className={styles.gained}>{followers}</span>
          <br />
        </>
      )}
      {contrib >= 0 && (
        <>
          <span className={styles.subName}>contrib</span>
          <span className={styles.gained}>{contrib}</span>
        </>
      )}
      {/* <br /> */}
      {/* <span dangerouslySetInnerHTML={{ __html: calendar }} /> */}
    </div>
  ) : (
    <></>
  );
};

export default GithubView;
