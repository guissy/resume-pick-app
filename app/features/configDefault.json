[
  {
    "name": "核心项",
    "children": [
      {
        "name": "RN",
        "alias": ["react native", "rn", "weex"],
        "score": 1.5
      },
      {
        "name": "flutter",
        "alias": ["flutter", "dart"],
        "score": 1.5
      },
      {
        "name": "h5",
        "alias": [
          "webview",
          "uni-app",
          "uniapp",
          "umi-app",
          "ionic",
          "apicloud",
          "mobile",
          "hybird"
        ],
        "score": 1.5
      },
      {
        "name": "react",
        "alias": ["redux", "antd", "ant design"],
        "score": 1.5
      },
      {
        "name": "vue",
        "alias": ["vue2", "vuejs", "vue2", "element", "vant"],
        "score": 1.5
      },
      {
        "name": "vue3",
        "score": 1.5
      },
      {
        "name": "angular",
        "alias": [
          "ng",
          "ng2",
          "angular2",
          "angular4",
          "angular5",
          "angular6",
          "angular7",
          "angular8"
        ],
        "score": 1
      },
      {
        "name": "wxapp",
        "alias": ["微信", "小程序", "taro", "mpvue", "mpvue"],
        "score": 1.5
      },
      {
        "name": "node",
        "alias": ["nodejs", "express", "keo", "egg", "next", "typeorm", "pm2"],
        "score": 1.5
      },
      {
        "name": "electron",
        "alias": ["nwjs"],
        "score": 1.5
      },
      {
        "name": "state mgr",
        "alias": [
          "redux",
          "saga",
          "mobx",
          "vuex",
          "vuetify",
          "dva",
          "事件总线",
          "eventBus",
          "event bus"
        ],
        "score": 1
      },
      {
        "name": "es6",
        "alias": [
          "es5",
          "es6+",
          "es2015",
          "es2016",
          "es2017",
          "es7",
          "es8",
          "es9",
          "ecmascript"
        ],
        "score": 0.5
      },
      {
        "name": "w3c",
        "score": 0.5,
        "alias": ["mdn"]
      },
      {
        "name": "css3",
        "alias": [
          "flex",
          "grid",
          "css3",
          "rem",
          "less",
          "sass",
          "css-modules",
          "css in js",
          "css-in-js",
          "styled-components",
          "jss",
          "emotion"
        ],
        "score": 0.5
      }
    ]
  },
  {
    "name": "进阶",
    "children": [
      {
        "name": "typescript",
        "alias": ["ts"],
        "score": 1.5
      },
      {
        "name": "rxjs",
        "alias": ["ReactiveX", "ngrx"],
        "score": 1.5
      },
      {
        "name": "graphql",
        "alias": ["gql"],
        "score": 1.5
      },
      {
        "name": "test",
        "calc": "max",
        "alias": [
          "mocha",
          "jasmine",
          "jest",
          "cypress",
          "puppeteer",
          "unit test"
        ],
        "score": 1.5
      },
      {
        "name": "lint",
        "alias": ["tslint", "eslint", "jslint", "命名规范", "代码规范"],
        "score": 0.5
      },
      {
        "name": "review",
        "alias": ["pr", "git flow", "代码审查"],
        "score": 0.5
      },
      {
        "name": "gayhub",
        "alias": ["github.com", "gitee.com"],
        "score": 1.5
      },
      {
        "name": "blog",
        "alias": [
          "个人博客",
          "博客地址",
          "csdn.net",
          "cnblogs.com",
          "juejin.im",
          "zhihu.com",
          "jianshu.com",
          "segmentfault.com",
          "oschina.net",
          "imooc.com",
          "51cto.com",
          "gitbooks.io"
        ],
        "score": 1.5
      }
    ]
  },
  {
    "name": "其他",
    "children": [
      {
        "name": "leader",
        "alias": ["组长", "负责人"],
        "score": 1
      },
      {
        "name": "mock",
        "score": 0.5
      },
      {
        "name": "git",
        "score": 0.5
      },
      {
        "name": "docker",
        "score": 0.5
      },
      {
        "name": "websocket",
        "alias": ["socket", "canvas", "charles"],
        "score": 0.5
      },
      {
        "name": "微服务",
        "score": 0.5
      },
      {
        "name": "加密",
        "alias": ["密文", "支付", "md5", "sha1", "签名", "白名单"],
        "score": 0.5
      },
      {
        "name": "重构",
        "alias": ["防抖", "去抖", "节流", "虚拟化", "懒加载"],
        "score": 0.5
      },
      {
        "name": "自定义",
        "alias": ["封装", "公共组件", "二次封装"],
        "score": 0.5
      },
      {
        "name": "python",
        "score": 0.5
      },
      {
        "name": "java",
        "score": 0.5
      },
      {
        "name": "go",
        "score": 0.5
      },
      {
        "name": "rust",
        "score": 0.5
      },
      {
        "name": "cpp",
        "score": 0.5
      },
      {
        "name": "swift",
        "score": 0.5
      },
      {
        "name": "kotlin",
        "score": 0.5
      },
      {
        "name": "scala",
        "score": 0.5
      },
      {
        "name": "mysql",
        "score": 0.5
      },
      {
        "name": "mongodb",
        "score": 0.5
      },
      {
        "name": "electron",
        "score": 0.5
      },
      {
        "name": "devops",
        "alias": ["shell", "nginx", "linux", "cicd"],
        "score": 0.5
      },
      {
        "name": "vim",
        "score": 0.5
      },
      {
        "name": "ssr",
        "score": 0.5
      },
      {
        "name": "bi",
        "score": 0.5,
        "alias": ["可视化", "大屏", "echarts", "d3"]
      },
      {
        "name": "three.js",
        "score": 0.5
      },
      {
        "name": "postman",
        "score": 0.5,
        "alias": ["charles", "fiddler"]
      },
      {
        "name": "leetcode",
        "score": 0.5,
        "alias": ["力扣", "leet code"]
      },
      {
        "name": "deno",
        "score": 0.5
      },
      {
        "name": "wasm",
        "score": 0.5,
        "alias": ["web assembly"]
      },
      {
        "name": "iframe",
        "alias": ["flash", "dreamweaver", "div+css", "v-for", "v-if"],
        "score": -1.5
      },
      {
        "name": "ajax",
        "alias": ["ajax", "jsonp", "ie6", "ie7", "ie8"],
        "score": -1
      },
      {
        "name": "jquery",
        "alias": ["jq", "zepto"],
        "score": -0.5
      },
      {
        "name": "svn",
        "alias": ["requirejs", "gulp", "grunt", "svn"],
        "score": -1
      },
      {
        "name": "积极乐观",
        "alias": ["为人和蔼", "吃苦耐劳", "上进", "了解"],
        "score": -0.5
      },
      {
        "name": "熟悉",
        "score": -0.3
      }
    ]
  }
]
