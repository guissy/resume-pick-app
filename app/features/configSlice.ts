import { createSlice } from '@reduxjs/toolkit';
// eslint-disable-next-line import/no-cycle
import { AppThunk, RootState } from '../store';
import { ConfigFile } from './type';
import initFile, { initDefaultFile, saveFile } from './configUtil';

export const configSlice = createSlice({
  name: 'config',
  initialState: { configs: {} as ConfigFile },
  reducers: {
    buildConfig: (state, action) => {
      if (Object.keys(state.configs).length === 0) {
        state.configs = action.payload;
      }
    },
    updateConfig: (state, action) => {
      Object.assign(state.configs, action.payload);
    },
    saveConfig: (state) => {
      saveFile(state.configs);
    },
    resetConfig: (state, action) => {
      state.configs = action.payload;
      saveFile(state.configs);
    },
  },
});

export const {
  updateConfig,
  buildConfig,
  saveConfig,
  resetConfig,
} = configSlice.actions;

export const initConfigAsync = (): AppThunk => (dispatch) =>
  initFile()
    .then((data) => dispatch(buildConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export const resetConfigAsync = (): AppThunk => (dispatch) =>
  initDefaultFile()
    .then((data) => dispatch(resetConfig(data)))
    // eslint-disable-next-line no-console
    .catch((e) => console.error(e));

export default configSlice.reducer;

export const selectConfig = (state: RootState) => state.config.configs.default;
