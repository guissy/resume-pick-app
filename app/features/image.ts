import RN_png from '../images/RN.png';
import react_png from '../images/react.png';
import angular_png from '../images/angular.png';
import css3_png from '../images/css3.png';
import docker_png from '../images/docker.png';
import es6_png from '../images/es6.png';
import flutter_png from '../images/flutter.png';
import gayhub_png from '../images/gayhub.png';
import electron_png from '../images/electron.png';
import gitee_png from '../images/gitee.png';
import git_png from '../images/git.png';
import go_png from '../images/go.png';
import graphql_png from '../images/graphql.png';
import hybird_png from '../images/hybird.png';
import java_png from '../images/java.png';
import kotlin_png from '../images/kotlin.png';
import lint_png from '../images/lint.png';
import linux_png from '../images/linux.png';
import mongodb_png from '../images/mongodb.png';
import mysql_png from '../images/mysql.png';
import nginx_png from '../images/nginx.png';
import node_png from '../images/node.png';
import python_png from '../images/python.png';
import test_png from '../images/test.png';
import typescript_png from '../images/typescript.png';
import vue_png from '../images/vue.png';
import wxapp_png from '../images/wxapp.png';
import zhihu_png from '../images/zhihu2.png';
import csdn_png from '../images/csdn.png';
import cnblogs_png from '../images/cnblogs.png';
import juejin_png from '../images/juejin.png';
import jianshu_png from '../images/jianshu.png';
import segmentfault_png from '../images/segmentfault.png';
import oschina_png from '../images/oschina.png';
import imooc_png from '../images/imooc.png';
import cto_png from '../images/cto.png';

export default {
  RN_png,
  react_png,
  angular_png,
  css3_png,
  docker_png,
  es6_png,
  flutter_png,
  gayhub_png,
  github_png: gayhub_png,
  git_png,
  go_png,
  graphql_png,
  hybird_png,
  java_png,
  kotlin_png,
  lint_png,
  linux_png,
  mongodb_png,
  mysql_png,
  nginx_png,
  node_png,
  python_png,
  test_png,
  typescript_png,
  vue_png,
  wxapp_png,
  zhihu_png,
  csdn_png,
  cnblogs_png,
  juejin_png,
  jianshu_png,
  segmentfault_png,
  oschina_png,
  imooc_png,
  cto_png,
  gitee_png,
  electron_png,
};
