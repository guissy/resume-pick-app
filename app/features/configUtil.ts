// eslint-disable-next-line @typescript-eslint/ban-ts-comment
// @ts-ignore
import storage from 'electron-storage';
import configDefault from './configDefault.json';
import { Config } from './type';

function initFile(): Promise<Record<string, Config>> {
  return storage.get('keywords.json').catch((err: unknown) => {
    // eslint-disable-next-line no-console
    console.error(err);
    return storage.set('keywords.json', { default: configDefault });
  });
}

export function initDefaultFile(): Promise<Record<string, Config>> {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return Promise.resolve({ default: configDefault });
}

export function saveFile(configs: Record<string, Config>) {
  return storage
    .set('keywords.json', JSON.stringify(configs, null, '  '))
    .catch((err: unknown) => {
      // eslint-disable-next-line no-console
      console.error(err);
    });
}

export default initFile;
