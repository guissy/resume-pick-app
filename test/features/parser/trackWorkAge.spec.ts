import mockScoreFile from '../utils/mockScoreFile';
import configDefault from '../../../app/constants/configDefault.json';
import trackWorkAge, {
  calcSentiment,
  getBlogByLink,
  getScoreMap,
  trackDegree,
  trackLinks,
  trackPhone,
  trackSalary,
  trackSchool,
} from '../../../app/features/parser/tractWorkAge';
import { Keyword } from '../../../app/features/type';

describe('trackWorkAge', () => {
  it('trackWorkAge', () => {
    expect(trackWorkAge('工作经验 7年')).toBe(7);
    expect(trackWorkAge('10年经验')).toBe(10);
  });
  it('trackPhone', () => {
    expect(trackPhone(`电话 1351234567890 `)).toBe('');
    expect(trackPhone(`电话 135-1234-5678 `)).toBe('13512345678');
    expect(trackPhone(`电话 135 1234 5678 `)).toBe('13512345678');
    Array(1000).forEach(() => {
      const p = (1e10 + 1e10 * Math.random()).toFixed(0);
      expect(trackPhone(`电话 ${p.toString()}`)).toBe(p.toString());
    });
  });
  it('trackSchool', () => {
    expect(trackSchool('负责华为大学业务')).toBeFalsy();
    expect(trackSchool('马云的湖畔学院')).toBeFalsy();
    expect(trackSchool('学院 深圳大学 2020')).toBe('深圳大学');
  });
  it('trackDegree', () => {
    expect(trackDegree('学历 ：博士')).toBe('博士');
    expect(trackDegree('学历 ：大专')).toBe('大专');
  });
  it('trackSalary', () => {
    expect(trackSalary('期望13-15k')).toBe('13-15k');
    expect(trackSalary('期望14K~22k')).toBe('14K~22k');
  });
  it('trackLinks', () => {
    expect(trackLinks('点击这里https://www.baidu.com/?q=3')[0]).toBe(
      'https://www.baidu.com/?q=3'
    );
  });
  it('getScoreMap', () => {
    expect(getScoreMap(mockScoreFile[0].keywords as Keyword[])).toMatchObject({
      vue: 1,
    });
    expect(getScoreMap(mockScoreFile[1].keywords as Keyword[])).toMatchObject({
      react: 1,
    });
  });
  it('calcSentiment', () => {
    expect(
      calcSentiment(mockScoreFile[0].text, configDefault as Keyword[])
    ).toBeGreaterThan(0);
  });
  it('getBlogByLink', () => {
    expect(getBlogByLink('https://github.com/guissy/cc')).toBe('github');
    expect(
      getBlogByLink('https://jiayili.gitbooks.io/fe-study-easier/content/')
    ).toBe('gitbooks');
    expect(getBlogByLink('https://blog.51cto.com/12131824')).toBe('51cto');
    expect(getBlogByLink('https://my.oschina.net/u/267941')).toBe('oschina');
    expect(getBlogByLink('https://www.imooc.com/u/1215284')).toBe('imooc');
    expect(getBlogByLink('https://segmentfault.com/u/jmxiao')).toBe(
      'segmentfault'
    );
    expect(getBlogByLink('https://www.jianshu.com/u/c7757daadf27')).toBe(
      'jianshu'
    );
    expect(getBlogByLink('https://www.zhihu.com/people/gui-zi-29/posts')).toBe(
      'zhihu'
    );
    expect(getBlogByLink('https://zhuanlan.zhihu.com/269840126')).toBe('zhihu');
    expect(getBlogByLink('https://blog.csdn.net/weixin_42134789')).toBe('csdn');
    expect(getBlogByLink('https://www.cnblogs.com/jiujuan/')).toBe('cnblogs');
    expect(getBlogByLink('https://juejin.im/user/3403743728515246/')).toBe(
      'juejin'
    );
  });
});
