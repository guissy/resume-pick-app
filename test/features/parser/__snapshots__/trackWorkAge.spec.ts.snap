// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trackWorkAge calcLevelSalaryRate 1`] = `
Object {
  "1": 0,
  "2": 0,
  "3": 0.5911330049261084,
  "4": 0.7881773399014779,
  "5": 0.9852216748768473,
  "6": 1.1822660098522169,
  "7": 1.3793103448275863,
  "8": 1.5763546798029557,
  "9": 1.7733990147783252,
}
`;

exports[`trackWorkAge calcLevelSalaryStar 1`] = `
Object {
  "0.2": -2.5,
  "0.4": 0,
  "0.6": 1.5,
  "0.8": 2.5,
  "1": 3.5,
  "1.2": 4,
  "1.4": 4.5,
  "1.6": 5,
  "1.8": 5.5,
  "2": 6,
  "2.2": 6,
}
`;
